icu4c/source/i18n/zonemeta.cpp

```diff
@@ -389,6 +390,35 @@ ZoneMeta::getCanonicalCLDRID(const TimeZone& tz) {
     return getCanonicalCLDRID(tz.getID(tzID), status);
 }
 
+UnicodeString& U_EXPORT2
+ZoneMeta::getIanaID(const UnicodeString& tzid, UnicodeString& ianaID, UErrorCode& status) {
+    // First, get CLDR canonical ID
+    const char16_t *canonicalID = getCanonicalCLDRID(tzid, status);
+    if (U_FAILURE(status) || canonicalID == nullptr) {
+        ianaID.setToBogus();
+        return ianaID;
+    }
+    // Find IANA mapping if any.
+    UErrorCode tmpStatus = U_ZERO_ERROR;
+    UnicodeString tmpKey(canonicalID);
+    tmpKey.findAndReplace(UnicodeString("/"), UnicodeString(":"));
+    char keyBuf[ZID_KEY_MAX + 1];
+    /* int32_t keyLen = */ tmpKey.extract(0, tmpKey.length(), keyBuf, sizeof(keyBuf), US_INV);
+
+    StackUResourceBundle r;
+    ures_openDirectFillIn(r.getAlias(), nullptr, gKeyTypeData, &tmpStatus);
+    ures_getByKey(r.getAlias(), gIanaMapTag, r.getAlias(), &tmpStatus);
+    ures_getByKey(r.getAlias(), gTimezoneTag, r.getAlias(), &tmpStatus);
+    int32_t tmpLen = 0;
+    const char16_t* tmpIana = ures_getStringByKey(r.getAlias(), keyBuf, &tmpLen, &tmpStatus);
+    if (U_SUCCESS(tmpStatus)) {
+        ianaID.setTo(tmpIana, tmpLen);
```

Comment:
since tmpIana is returned by ures_getStringByKey that is "a pointer to a zero-terminated UChar array which lives in a memory mapped/DLL file." Should we use the  ianaID.setTo(true, tmpIana, tmpLen);
to save some memory?
https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/classicu_1_1UnicodeString.html#a9a2f115fb0975766d9b8823d98622295

Reply:
Thanks for the suggestion. Will do.

Reply:
Good idea